mlflow/utils/conda.py

```diff
@@ -84,12 +95,58 @@ def _get_conda_executable_for_create_env():
     return conda_env_create_path
 
 
-def _list_conda_environments():
-    prc = process._exec_cmd([get_conda_bin_executable("conda"), "env", "list", "--json"])
+def _list_conda_environments(conda_extra_envs=None):
+    """
+    Return a list of names of conda environments.
+
+    :param conda_extra_envs: extra environment variables generated by `_get_conda_extra_envs`.
```

Comment:
`_get_conda_extra_envs` doesn't seem to exist.

Reply:
Does `conda_extra_envs` has to be created by `_get_conda_extra_envs`?

Reply:
no. updated code